src/types.rs

```diff
@@ -248,6 +248,66 @@ impl From<JsonValue> for Value {
     }
 }
 
+/// Convert Avro values to Json values
+impl std::convert::TryFrom<Value> for JsonValue {
+    type Error = crate::errors::Error;
+    fn try_from(value: Value) -> AvroResult<Self> {
+        match value {
+            Value::Null => Ok(JsonValue::Null),
+            Value::Boolean(b) => Ok(JsonValue::Bool(b)),
+            Value::Int(i) => Ok(JsonValue::Number(i.into())),
+            Value::Long(l) => Ok(JsonValue::Number(l.into())),
+            Value::Float(f) => Number::from_f64(f.into())
+                .map(JsonValue::Number)
+                .ok_or_else(|| Error::JsonTryFrom("failed to convert f64 to json".to_string())),
+            Value::Double(d) => Number::from_f64(d)
+                .map(JsonValue::Number)
+                .ok_or_else(|| Error::JsonTryFrom("failed to convert f64 to json".to_string())),
+            Value::Bytes(bytes) => Ok(JsonValue::Array(
+                bytes.into_iter().map(|b| b.into()).collect(),
+            )),
+            Value::String(s) => Ok(JsonValue::String(s)),
+            Value::Fixed(_size, items) => Ok(JsonValue::Array(
+                items.into_iter().map(|v| v.into()).collect(),
+            )),
+            Value::Enum(_i, s) => Ok(JsonValue::String(s)),
+            Value::Union(b) => JsonValue::try_from(*b),
+            Value::Array(items) => {
+                let results: Result<Vec<JsonValue>, Error> =
+                    items.into_iter().map(JsonValue::try_from).collect();
+                results.map(JsonValue::Array)
```

Comment:
Can you write this as `Ok(Self::Array(items.into_iter().map(Self::try_from).collect::<Result<Vec<_>, _>>()?))` and avoid the type declaration?

Reply:
+1